### PR TITLE
optimize kube informers wait cache sync

### PR DESCRIPTION
**Pull request by hzxuzhonghu**

### CODE REVIEW

File: pilot/pkg/serviceregistry/kube/controller/controller.go [hunk 1]
```diff
@@ -477,23 +464,13 @@ func (c *Controller) Run(stop <-chan struct{}) {
...
-		c.queue.Run(stop)
-	}()
-
-	go c.services.Run(stop)
+	go c.serviceInformer.Run(stop)
 	go c.pods.informer.Run(stop)
 	go c.nodeMetadataInformer.Run(stop)
 	go c.filteredNodeInformer.Run(stop)
-
-	// To avoid endpoints without labels or ports, wait for sync.
```
Comment (howardjohn) [inline pilot/pkg/serviceregistry/kube/controller/controller.go:491]: this seems intentionally before endpoints, any reason its safe to remove now?

File: pilot/pkg/serviceregistry/kube/controller/controller.go [hunk 2]
```diff
@@ -270,18 +271,7 @@ func (c *Controller) Cluster() string {
...
 
-func (c *Controller) checkReadyForEvents() error {
-	if !c.HasSynced() {
-		return errors.New("waiting till full synchronization")
-	}
-	return nil
-}
-
 func (c *Controller) onServiceEvent(curr interface{}, event model.Event) error {
-	if err := c.checkReadyForEvents(); err != nil {
```
Comment (howardjohn) [inline pilot/pkg/serviceregistry/kube/controller/controller.go:281]: why do we not need this?

Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/kube/controller/controller.go:281]: Before queue run, we have waited cache sync. That is same as what `checkReadyForEvents` does

File: pilot/pkg/serviceregistry/kube/controller/controller.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/kube/controller/controller.go:491]: As we can see, the actual handler will be executed after cache sync in line 483, so it is same now
https://github.com/istio/istio/pull/23745/files#diff-288be3f23828adceb9ebb059ce53a23eL483

**Pull request closed without merging**